lib/WasmReader/WasmSignature.cpp

```diff
@@ -181,53 +181,31 @@ WasmSignature::FromIDL(WasmSignatureIDL* sig)
     return reinterpret_cast<WasmSignature*>(sig);
 }
 
-void
-WasmSignature::Dump()
+uint32
+WasmSignature::WriteSignatureToString(_Outptr_opt_result_buffer_(maxlen) char16* out, uint32 maxlen)
 {
-#ifdef ENABLE_DEBUG_CONFIG_OPTIONS
-    Output::Print(_u("("));
-    for(uint32 i = 0; i < this->GetParamCount(); i++)
+    AssertOrFailFast(out != nullptr);
+    uint32 numwritten = 0;
+    numwritten += swprintf_s(out+numwritten, maxlen-numwritten, _u("("));
+    for (uint i = 0; i < this->GetParamCount(); i++)
     {
-        if(i != 0)
+        if (i != 0)
         {
-            Output::Print(_u(", "));
-        }
-        switch(this->GetParam(i)) {
-            case WasmTypes::WasmType::Void:
-                Output::Print(_u("void"));
-                break;
-            case WasmTypes::WasmType::I32:
-                Output::Print(_u("i32"));
-                break;
-            case WasmTypes::WasmType::I64:
-                Output::Print(_u("i64"));
-                break;
-            case WasmTypes::WasmType::F32:
-                Output::Print(_u("f32"));
-                break;
-            case WasmTypes::WasmType::F64:
-                Output::Print(_u("f64"));
-                break;
+            numwritten += swprintf_s(out+numwritten, maxlen-numwritten, _u(", "));
         }
+        numwritten += swprintf_s(out+numwritten, maxlen-numwritten, _u("%ls"), WasmTypes::GetTypeName(this->GetParam(i)));
```

Comment:
>numwritten += swprintf_s(out+numwritten, maxlen-numwritten, _u("%ls"), WasmTypes::GetTypeName(this->GetParam(i))); [](start = 8, length = 114)

Add another test where the combined string length of parameters is more than 511